crates/nu-parser/src/parser.rs

```diff
@@ -2449,6 +2453,94 @@ pub fn unescape_string(bytes: &[u8], span: Span) -> (Vec<u8>, Option<ParseError>
                     }
                     idx += 5;
                 }
+                Some(b'x') => {
+                    match (bytes.get(idx + 1), bytes.get(idx + 2)) {
+                        (Some(h1), Some(h2)) => {
+                            let s = String::from_utf8(vec![*h1, *h2]);
+
+                            if let Ok(s) = s {
+                                let int = u32::from_str_radix(&s, 16);
+
+                                if let Ok(int) = int {
+                                    let result = char::from_u32(int);
+
+                                    if let Some(result) = result {
+                                        let mut buffer = vec![0; 4];
+                                        let result = result.encode_utf8(&mut buffer);
+
+                                        for elem in result.bytes() {
+                                            output.push(elem);
+                                        }
+
+                                        idx += 3;
+                                        continue;
+                                    }
```

Comment:
Should it be explicitly reported if it is not a valid single byte utf-8 char and thus ignored as nu is living in the utf-8 world? 
I think I can live with the trade-off that it is impossible like this to create a multi-byte unicode character with the binary/byte escapes as the `"\uXXXX"` exists

Reply:
This is a good point, one use case of this is to construct potentially non-valid utf-8 for sending escape sequences to terminals. Maybe Binary is a better option for this? In which case I should back out the string changes. This was the primary problem I was looking to solve:

```
❯ ^printf "\163"
Error: nu::parser::parse_mismatch (link)

  × Parse mismatch during operation.
   ╭─[entry #12:1:1]
 1 │ ^printf "\163"
   ·          ──┬──
   ·            ╰── expected supported escape character
   ╰────
```

And maybe it is good enough that I can just do:

```
❯ ^printf "\\163"
s
```

Reply:
I only have seen the seven-bit ANSI escape sequences in the wild that use valid ASCII and have only the lower seven bits set and are thus valid UTF-8 (multibyte UTF-8 encoding has the most significant bit set). The ANSI/ECMA spec allows for the 8-bit "C1" codes but this is UTF-8 incompatible and replaced with "C0" and encoding.

The octal code in your example is a lowercase `s` in my book. 

BTW for your printing without newlines usecase there is `print -n` in the nushell world as well as the `ansi` family of commands with names for the common ones and utilities to send the `ESC` or the `CSI`.

Reply:
Is `print -n` an unreleased flag?

```
❯ print -n "hello"
Error: nu::parser::unknown_flag (link)

  × The `print` command doesn't have flag `-n`.
```

Reply:
Yeah has been added since the last release https://github.com/nushell/nushell/pull/5458

Reply:
Cool, looking forward to it. As for the escape support for `\x` and `\o`, I am backing that change out.